fe/lib/components/JobCategorySelect/JobCategorySelect.tsx

```diff
@@ -58,6 +66,39 @@ export const JobCategorySelect = forwardRef<HTMLInputElement, Props>(
     });
 
     const [selectedJobCategoryId, setSelectedJobCategoryId] = useState('');
+    const [initialJobCategory, setInitialJobCategory] = useState<
+      JobCategoryAttributesFragment | undefined
+    >();
+
+    const loadInitialJobCategory = useCallback(async () => {
+      if (!initialValue || initialJobCategory?.id.value === initialValue) {
+        return;
+      }
+
+      if (!client) {
+        return;
+      }
+      const { data } = await client.query({
```

Comment:
Do we need to query the SEEK API again? Aren't we already querying the entire hierarchy with `JOB_CATEGORIES`?

Reply:
Would something like this work in `JobCategorySelectInput` instead:

```typescript
useEffect(() => {
  if (initialJobCategoryId && selectedParentCategoryId === '') {
    for (const parentCategory of jobCategories) {
      if (!parentCategory.children) {
        // Shouldn't happen but TypeScript is suspicious
        continue;
      }

      if (parentCategory.id === initialJobCategoryId) {
        // Passed a parent category ID
        setChildCategories(parentCategory.children);
        setSelectedParentCategoryId(parentCategory.id.value);
        return;
      }

      for (const childCategory of parentCategory.children) {
        if (childCategory.id === initialJobCategoryId) {
          setChildCategories(parentCategory.children);
          setSelectedParentCategoryId(parentCategory.id.value);
          setSelectedChildCategoryId(parentCategory.id.value);
          return;
        }
      }
    }
  }
}, [initialJobCategoryId, jobCategories, selectedParentCategoryId]);
```

Reply:
Hmm, I was trying to avoid a nested loop. But nested loop vs querying the api, it makes sense to have the nested loop.